src/db.c

```diff
@@ -2838,9 +2841,9 @@ int bitfieldGetKeys(struct serverCommand *cmd, robj **argv, int argc, getKeysRes
     return 1;
 }
 
-bool dbHasNoKeys(void) {
+bool dbsHaveNoKeys(void) {
```

Comment:
It's odd that this function is in db.c while dbHasNoKeys and createDatabase is in server.c.

I like to think of server.c as the unit handling the global server state and db.c handling the serverDb abstract datatype. This is not true though. The abstractions are mixed and the globals are accessed from all over the place.

Anyway, we can try to avoid making this mess even more messy by keeping these related functions together, or do you have another idea?

Reply:
I've moved it to server.c (dbHasNoKeys was originally in db.c). Now both are in server.c

Reply:
OK. What's your reasoning to prefer server.c vs db.c for these?

I remember in the multi-db PR, someone suggested to move dbHasNoKeys to db.c. There was not clear explanation though, maybe only based on the function name...

Note: I don't really prefer either one of then, only that similar/related ones are together.

Reply:
IMO, logically `db.c` shouldn't know about server.c. Both `dbHasNoKeys` and `dbsHaveNoKeys` access a global state owned by `server.c`, so conceptually they may belong there. But in practice, it's intertwined... Either is fine.. 

This function for example
```
/* Return non-zero if the database is empty  */
int dbHasNoKeys(int dbid) {
    return dbid < 0 || dbid >= server.dbnum || !server.db[dbid] || kvstoreSize(server.db[dbid]->keys) == 0;
}
```
Knows about server.db, and dbnum, and dbid.. These concepts belong in server.c

Reply:
Thanks. Now we have this explanation spelled out in case someone suggests move it again. :)

Just a though experiment: If we ever want to clean up these abstractions to the point where only server.c has access to the server struct and only db.c has access to the serverDb struct, then we may need to split these in two: `dbHasNoKeys(serverDb *db);` in db.c and `serverDbHasNoKeys(int dbid);` in server.c that calls the other one. That's not going to happen any day soon though.